packages/bitcoin/src/wallet/lib/tx-builder/utils.ts

```diff
@@ -0,0 +1,73 @@
+import * as bitcoin from 'bitcoinjs-lib';
+import { InputResolver, TransactionHistoryEntry, TransactionStatus } from '../providers';
+import { Network } from '../common';
+
+/**
+ * Converts a raw transaction hexadecimal string into a TransactionHistoryEntry.
+ *
+ * @param {string} raw - The raw transaction hexadecimal string.
+ * @param {Network} network - The network type (Mainnet or Testnet) to which the transaction belongs.
+ * @param inputResolver - The input resolver.
+ * @returns {TransactionHistoryEntry} The structured transaction history entry derived from the raw transaction.
+ */
+/**
+ * Converts a raw transaction hex into a TransactionHistoryEntry using an InputResolver to fetch additional input details.
+ *
+ * This function deserializes a raw Bitcoin transaction hex and transforms it into a transaction history entry.
+ * It uses an InputResolver to asynchronously fetch and include details about each input, such as the associated address
+ * and the amount of satoshis, which are not directly available from the transaction input script.
+ *
+ * @param {string} raw - The raw hexadecimal string of the Bitcoin transaction.
+ * @param {Network} network - The Bitcoin network (Mainnet or Testnet) that the transaction pertains to.
+ * @param {InputResolver} inputResolver - An instance of InputResolver to fetch additional input details.
+ * @returns {Promise<TransactionHistoryEntry>} A promise that resolves to a detailed transaction history entry.
+ */
+export const historyEntryFromRawTx = async (
+  raw: string,
+  network: Network,
+  inputResolver: InputResolver
+): Promise<TransactionHistoryEntry> => {
+  const tx = bitcoin.Transaction.fromHex(raw);
+  const bitcoinNetwork = network === Network.Mainnet ? bitcoin.networks.bitcoin : bitcoin.networks.testnet;
+
+  const resolvedInputs = await Promise.all(
+    tx.ins.map(async (input) => {
+      const txId = Buffer.from(input.hash).reverse().toString('hex');
+      const index = input.index;
+      try {
+        const resolvedInput = await inputResolver.resolve(txId, index);
+        return {
+          ...resolvedInput,
+          txId,
+          index
+        };
```

Comment:
```suggestion
        return resolvedInput;
```

Aren't `txId` and `index` already present in `resolvedInput`?

Reply:
done here https://github.com/input-output-hk/lace/pull/1809/commits/6e68df7521323a3a448a06d91d519c8afe292a26